eventsourcing/src/main/java/org/axonframework/eventsourcing/annotations/AnnotatedEventSourcingComponent.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.annotations;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.eventhandling.EventHandler;
+import org.axonframework.eventhandling.EventHandlerRegistry;
+import org.axonframework.eventhandling.EventHandlingComponent;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventsourcing.EventSourcingComponent;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.QualifiedName;
+import org.axonframework.messaging.annotation.AnnotatedHandlerInspector;
+import org.axonframework.messaging.annotation.ClasspathHandlerDefinition;
+import org.axonframework.messaging.annotation.ClasspathParameterResolverFactory;
+import org.axonframework.messaging.annotation.HandlerDefinition;
+import org.axonframework.messaging.annotation.MessageHandlingMember;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+// todo: maybe extends AnnotatedEventHandlingComponent? Implement SimpleEventSourcingComponent
+public class AnnotatedEventSourcingComponent<T> implements EventSourcingComponent {
```

Comment:
Works same as `AnnotatedEventHandlingComponent`, just allows to return Single for the `@EventHandler` or `@EventSourcingHandler`.

Reply:
Amazing! :-)